### PR TITLE
Fixes #30185 - Change console files to be compatible to python3

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

**Comment by ekohl** [comment]
I wonder if we should put in the effort to unbundle [websockify](https://github.com/novnc/websockify) and use it as an external dependency.

**Comment by shiramax** [comment]
@ekohl, do you know what's the reason to not do it in the first place?

**Comment by ekohl** [comment]
Historical reasons I guess. That part of the codebase has been mostly untouched for years now and the vendoring is even older. When speaking to @ares in a related conversation he didn't even know it was an external project and assumed someone wrote it specifically for Foreman. I created https://projects.theforeman.org/issues/29539 and https://community.theforeman.org/t/consolidating-the-console/11215 spoke about the refactor even a step further.

In theory it would be:
* Check whether Websockify 0.8.0 is sufficient (which is in EPEL, Debian 10 and Ubuntu 18.04) or if we need 0.9.0 (https://bugzilla.redhat.com/show_bug.cgi?id=1628636 / https://bugzilla.redhat.com/show_bug.cgi?id=1816608)
* Modify the Foreman to search in $PATH (using `Foreman::Util::which`)
https://github.com/theforeman/foreman/blob/231761f7c62bd82e02a33ba184ccee102f9b9e57/lib/ws_proxy.rb#L67-L69
* Remove vendored code
* Change packaging to depend on python(3)-websockify
* Check whether SELinux is still happy (@lzap indicated this might be an issue and was willing to help)

Of course in practice you may find more things, but the general outline is pretty straight forward.

cdd8424fb4d31b390531cc1a1d74fb3820bae31a says it updated to 1.6.1 but this version doesn't exist. It may refer to 0.6.1 in which case 0.8.0 is actually an upgrade. In the past websockify lived in the noVNC repository but that also doesn't have a 1.6.1 version. Maybe @lzap remembers.

I probably don't have time for this, but I'll gladly help anyone who picks this up.

**Comment by shiramax** [comment]
@ekohl I can have a look, should I drop this one?

**Comment by ekohl** [comment]
@shiramax let's give it a try. If it doesn't work, we can circle back to this.

**Comment by shiramax** [comment]
@ekohl I decided to re-open this issue, because I don't have time at the moment to unbundle websockify and I think it would be best to add this PR for now.
I do plan to check websockify unbundle, but I don't have the time at the moment, and I don't know when I will have the time.

File: extras/noVNC/websockify/websocketproxy.py [hunk 1]
```diff
@@ -17,14 +17,14 @@
 try:    from http.server import HTTPServer
 except: from BaseHTTPServer import HTTPServer
 from select import select
-import websocket
+from .websocket import WebSocketRequestHandler, WebSocketServer, ssl
```
Comment (ekohl) [inline extras/noVNC/websockify/websocketproxy.py]: This does break if you execute this file directly:
```
$ ./extras/noVNC/websockify/websocketproxy.py --help
Traceback (most recent call last):
  File "./extras/noVNC/websockify/websocketproxy.py", line 20, in <module>
    from .websocket import WebSocketRequestHandler, WebSocketServer, ssl
ImportError: attempted relative import with no known parent package
```

However, we don't use that so we don't really care.

I would like to keep the patch small:
```suggestion
from . import websocket
```

Then you can revert the other changes in this file

File: extras/noVNC/websockify/__init__.py [hunk 2]
```diff
@@ -1,2 +1 @@
-from websocket import *
-from websocketproxy import *
+from .websocketproxy import websockify_init
```
Comment (ekohl) [inline extras/noVNC/websockify/__init__.py]: I'd prefer to keep the API the same and only change the bare minimum.
```suggestion
from .websocket import *
from .websocketproxy import *
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: extras/noVNC/websockify/__init__.py [hunk 3]
```diff
@@ -1,2 +1,2 @@
-from websocket import *
-from websocketproxy import *
+from .websocket import *
+from .websocketproxy import *
```
Comment (ekohl) [inline extras/noVNC/websockify/__init__.py]: Mind keeping the line ending?

Comment (shiramax) [reply extras/noVNC/websockify/__init__.py]: done :)

**Review by ekohl — APPROVED** [review]
I don't think we need CI to complete since there's no coverage for this anyway. I manually verified it using `websockify.py --help` and that worked.

**Pull request merged**